examples/pipeline_tacotron2/text/text_preprocessing.py

```diff
@@ -0,0 +1,60 @@
+"""Modified from https://github.com/keithito/tacotron"""
+
+from typing import List
+import re
+from unidecode import unidecode
+
+from .numbers import normalize_numbers
+
+
+# Regular expression matching whitespace:
+_whitespace_re = re.compile(r'\s+')
+
+# List of (regular expression, replacement) pairs for abbreviations:
+_abbreviations = [(re.compile('\\b%s\\.' % x[0], re.IGNORECASE), x[1]) for x in [
+    ('mrs', 'misess'),
+    ('mr', 'mister'),
+    ('dr', 'doctor'),
+    ('st', 'saint'),
+    ('co', 'company'),
+    ('jr', 'junior'),
+    ('maj', 'major'),
+    ('gen', 'general'),
+    ('drs', 'doctors'),
+    ('rev', 'reverend'),
+    ('lt', 'lieutenant'),
+    ('hon', 'honorable'),
+    ('sgt', 'sergeant'),
+    ('capt', 'captain'),
+    ('esq', 'esquire'),
+    ('ltd', 'limited'),
+    ('col', 'colonel'),
+    ('ft', 'fort'),
+]]
+
+_pad        = '_'
+_punctuation = '!\'(),.:;? '
+_special = '-'
+_letters = 'abcdefghijklmnopqrstuvwxyz'
+
+symbols = [_pad] + list(_special) + list(_punctuation) + list(_letters)
+_symbol_to_id = {s: i for i, s in enumerate(symbols)}
+
+
+def text_to_sequence(sent: str) -> List[int]:
+    r'''Converts a string of text to a sequence of IDs corresponding to the symbols in the text.
+
+      Args:
+        sent (str): The input sentence to convert to a sequence.
+
+      Returns:
+        List of integers corresponding to the symbols in the sentence.
+    '''
+    sent = unidecode(sent)  # convert to ascii
+    sent = sent.lower()  # lower case
+    sent = normalize_numbers(sent)  # expand numbers
+    for regex, replacement in _abbreviations:  # expand abbreviations
+        sent = re.sub(regex, replacement, sent)
+    sent = re.sub(_whitespace_re, ' ', sent)  # collapse whitespace
+
+    return [_symbol_to_id[s] for s in sent if s in _symbol_to_id]
```

Comment:
So the preprocessing will ignore the characters that are not in ``symbol_to_id``. Where is the preprocessing used in tacotron?

Reply:
Yes, it will ignore characters that are not in it. You can only encode a finite set of symbols. This preprocessing will be used to preprocess the input text before sending into Tacotron2. You want to encode each character into a number.

Reply:
I see. I guess it's only used in tacotron2 since other TTS model focus on the vocoder. If so, we can just put it here, otherwise we can put it as a general TTS function.

Reply:
Yes, for now it is only for Tacotron2. But I doubt this will go into the core library as this is a text preprocessing function and torchaudio is an audio/signal processing library.

Reply:
It can be a general function if it's a standard processing method in TTS. Since it's Tacotron specific let's keep it here :)